pkg/client/client.go

```diff
@@ -34,6 +37,44 @@ func decorateConfig(config *rest.Config, ua string) *rest.Config {
 	return cp
 }
 
+func BuildConfigFromClusterAndSecret(cluster *shipper.Cluster, secret *corev1.Secret) *rest.Config {
+	config := &rest.Config{
+		Host: cluster.Spec.APIMaster,
+	}
+
+	_, tokenOK := secret.Data["token"]
+	if tokenOK {
+		ca := secret.Data["ca.crt"]
+		config.CAData = ca
+
+		token := secret.Data["token"]
+		config.BearerToken = string(token)
+		return config
+	}
+
+	// Let's figure it's either a TLS secret or an opaque thing formatted
+	// like a TLS secret.
+	if ca, ok := secret.Data["tls.ca"]; ok {
```

Comment:
CA bundle comes in base64 encoding. I wonder if we have to do an explicit decoding before feeding this to the config?

Reply:
From the docs:

> CAData holds PEM-encoded bytes (typically read from a root certificates bundle).

Further, it seems like PEM-encoded bytes essentially means base64 (https://golang.org/src/encoding/pem/pem.go):

```
// The encoded form is:
//    -----BEGIN Type-----
//    Headers
//    base64-encoded Bytes
//    -----END Type-----
// where Headers is a possibly empty sequence of Key: Value lines.
```

So I read that this means no decoding is necessary, the CAData is expected to be in base64 already.

Reply:
Great, thanks for the clarification!